### PR TITLE
Fix propertySelect not disabling when no properties were available(2.1 cherry-pick)

### PR DESCRIPTION
**Pull request by joeybrk372**
- [x] @joeferner
- [x] @srfarley @kunklejr
- [x] @mwizeman @dsingley @EvanOxfeld 
- [x] @joeybrk372 @sfeng88 @rygim @jharwig 

Also fix i property search

CHANGELOG
Fixed: Property select field not disabling properly when there are no available properties.

### CODE REVIEW

File: web/war/src/main/webapp/js/util/ontology/propertySelect.js [hunk 1]
```diff
@@ -265,7 +232,57 @@ define([
...
+            this.select('findPropertySelection')
+                .attr('placeholder', i18n(placeholderMessage))
+                .attr('disabled', !hasProperties);
+        };
+
+        this.filterSourceProperties = function(properties) {
+            var self = this;
+
+            return _.chain(properties)
+                .filter(function(p) {
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/util/ontology/propertySelect.js]: As `filter` is the only step in the chain, assuming `properties` is an array, I'd recommend using `Array.prototype.filter` over `_.chain` and/or `_.filter`.

File: web/war/src/main/webapp/js/util/ontology/propertySelect.js [hunk 2]
```diff
@@ -76,66 +75,32 @@ define([
...
+                                .sortBy(function(itemJson) {
+                                    var item = JSON.parse(itemJson),
+                                        lower = item.displayName.toLowerCase();
+
+                                    if (item.propertyGroup) {
+                                        return '1' + item.propertyGroup + lower;
+                                    }
+                                    return '0' + lower;
+                                })
+                                .value();
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/util/ontology/propertySelect.js]: I like how `filterSourceProperties` is broken out into its own function above. I'd recommend refactoring one step further using a more declarative approach by splitting out the filter, map, and sortBy functions.

For example:

```
return _.chain(self.propertiesForSource)
  .filter(filterBySourceProperties)
  .map(mapProperty)
  .sortBy(sortByPropertyGroupAndDisplayName)
  .value();
```

File: web/war/src/main/webapp/js/util/ontology/propertySelect.js [hunk 3]
```diff
@@ -265,7 +236,57 @@ define([
...
+            this.select('findPropertySelection')
+                .attr('placeholder', i18n(placeholderMessage))
+                .attr('disabled', !hasProperties);
+        };
+
+        this.filterSourceProperties = function(properties) {
+            var self = this;
+
+            return _.chain(properties)
+                .filter(function(p) {
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/util/ontology/propertySelect.js]: This `filter` is the only item in the chain, which is what led to my suggestion to refactor `filterSourceProperties` back into the `typeahead`'s `source` function.

Alternatively, I recommend using `Array.prototype.filter` over a single item `_.chain`. `_.chain` adds overhead https://medium.com/making-internets/why-using-chain-is-a-mistake-9bc1f80d51ba#.2qttwkyck

Comment (jharwig) [inline web/war/src/main/webapp/js/util/ontology/propertySelect.js]: 👍 Prefer `Array.prototype.map` and `filter` if only one step.

Eventually we could move to transducers that alleviate the multiple passes through the array using `Array.prototype.reduce` and move all the way to FP.

https://medium.com/@roman01la/understanding-transducers-in-javascript-3500d3bd9624#.ziildwllq

**Pull request merged**